doc/source/reference/demeshening.rst

```diff
@@ -0,0 +1,161 @@
+
+In brief, however, what this relies on is two numbers, ``index_order1`` and
+``index_order2``.  These control the "coarse" and "refined" sets of indices,
+and they are supplied to any particle dataset ``load()`` in the form of a tuple
+as the argument ``index_order``.  By default these are set to 5 and 7,
+respectively, but it is entirely likely that a different set of values will
+work better for your purposes.
+
+For example, if you were to use the sample Gadget-3 dataset, you could override
+the default values and use values of 5 and 5 by specifying this argument to the
+``load_sample`` function; this works with ``load`` as well.
+
+.. code-block:: python
+
+   ds = yt.load_sample("Gadget3-snap-format2", index_order=(5, 5))
+
+So this is how you *change* the index order, but it doesn't explain precisely
+what this "index order" actually is.
+
+Index Order
+-----------
+
+The bitmap index system is based on a two-level scheme for assigning positions
+in three-dimensional space to integer values.  What this means is that each
+particle is assigned a "coarse" index, which is global to the full domain of
+the collection of particles, and *if necessary* an additional "refined" index
+is assigned to the particle, within that coarse index.
+
+The index "order" values refer to the number of entries on a side that each
+index system is allowed.  For instance, if we allow the particles to be
+subdivided into 8 "bins" in each direction, this would correspond to an index
+order of 3 (as $2^3 = 8$); correspondingly, an index order of 5 would be 32
+bins in each direction, and an index order of 7 would be 128 bins in each
+direction.  Each particle is then assigned a set of i, j, k values for the bin
+value in each dimension, and these i, j, k values are combined into a single
+(64-bit) integer according to a space-filling curve.
+
+The process by which this is done by yt is as follows:
+
+  1. For each "chunk" of data -- which may be a file, or a subset of a file in
+     which particles are contained -- assign each particle to an integer value
+     according to the space-filling curve and the coarse index order.  Set the
+     "bit" in an array of boolean values that each of these integers correspond
+     to.  Note that this is almost certainly *reductive* -- there will be fewer
+     bits set than there are particles, which is *by design*.
+  2. Once all chunks or files have been assigned an array of bits that
+     correspond to the places where, according to the coarse indexing scheme,
+     they have particles, identify all those "bits" that have been set by more
+     than one chunk.  All of these bits correspond to locations where more than
+     one file contains particles -- so if you want to select something from
+     this region, you'd need to read more than one file.
+  3. For each "collision" location, apply a *second-order* index, to identify
+     which sub-regions are touched by more than one file.
```

Comment:
You're referencing a second file here, and I'm not sure what you mean?  A second chunk? It seems like any time you have more particles than there are coarse bins available, you're going to trigger this step, regardless of if the dataset is broken into multiple files.  But perhaps I'm misunderstanding.  In re-reading this a few times, I think it may be clearer if you continuing to define things as chunks instead of jumping between files and chunks.

Reply:
I'll see if I can clear this up with a revision to the text, but no, you will not always trigger this.

Imagine you have two files (or chunks in those files -- it doesn't actually matter, as the sub-chunk inside a file will be the atomic unit) and in the first one, it contains only particles between [0.0, 0.0, 0.0] and [1.0, 1.0, 0.5], and in file two, the particles are all [0.0, 0.0, 0.5], [1.0, 1.0, 1.0] .  There would be no collisions between these two.

Now imagine that they aren't as cleanly load balanced as that, and you have some mixing.  In those bins where both files "touch" (by means of a particle) a bin, the second phase of refined indexing will be triggered.